index.js

```diff
@@ -0,0 +1,86 @@
+const core = require('@actions/core');
+const aws = require('aws-sdk');
+const fs = require('fs');
+const tmp = require('tmp');
+const REGION = process.env.AWS_REGION;
+const SERVICE = process.env.SERVICE;
+const PROVIDED_IMAGE_TAG = core.getInput('provided-image-tag');
+const DOCKER_TAG = process.env.DOCKER_TAG;
+const ssm = new aws.SSM({region: REGION});
+
+async function determineImageTag() {    
+    core.setOutput('image-tag', PROVIDED_IMAGE_TAG == '' ? DOCKER_TAG : PROVIDED_IMAGE_TAG);
+}
+
+async function getTaskDefinition() {
+    let ecs = new aws.ECS({region: REGION});
+    let output = await ecs.describeTaskDefinition({ 
+        taskDefinition: process.env.ECS_TASK_DEFINITION
+    }).promise();
+
+    return output.taskDefinition;
+}
+
+async function getParameters(path = `/ecs/${SERVICE}`, memo = [], nextToken) {
+    return ssm
+        .getParametersByPath({ Path: path, NextToken: nextToken, MaxResults: 10 })
+        .promise()
+        .then(({ Parameters, NextToken }) => {
+            const newMemo = memo.concat(Parameters);
+            return NextToken ? getParameters(path, newMemo, NextToken) : newMemo;
+        });
+}
+
+async function parseParameters(params) {
+    const arns = params.map(x => x.ARN);
+    
+    return arns.map(function (arn) {
+        let parts = arn.split('/');
+        let name = parts.at(-1);
+
+        return { 'valueFrom': arn, 'name': name }
+    });
+}
+
+async function updateTaskDefinition(taskDefinition, parameters) {    
+    let selectedDefinition = taskDefinition.containerDefinitions.filter(function(container) { 
+        return container.name === 'app'; 
+    })[0];
+
+    if (!selectedDefinition) {
+        throw new Error('Unable to find task definition container with name: app');
+    }
+
+    selectedDefinition.secrets = parameters
+    
+    return taskDefinition;
+}
+
+async function writeNewTaskDefinition(taskDefinition) {
+    var updatedTaskDefFile = tmp.fileSync({
+      tmpdir: process.env.RUNNER_TEMP,
+      prefix: 'task-definition-',
+      postfix: '.json',
+      keep: true,
+      discardDescriptor: true
+    });
+    const newTaskDefContents = JSON.stringify(taskDefinition, null, 2);
+    fs.writeFileSync(updatedTaskDefFile.name, newTaskDefContents);
+    core.setOutput('task-definition', updatedTaskDefFile.name);
+    return newTaskDefContents;
+}
+
+async function run() {
+    try {
+        await determineImageTag();
+        const taskDefinition = await getTaskDefinition();
```

Comment:
```suggestion
        const taskDefinition = await getTaskDefinition(service_name);
```

Feels a bit more well-encapsulated if we deal with the messy outside world outside of the `getTaskDefinition` function perhaps, then pass it what it needs to know as an argument?